File: 2. Medium/Java/Integer to Roman.java
class Solution {
    public String intToRoman(int num) {
        String[] letters = {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};
        int[] values = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
        String roman = "";
        int quotient = 0;
        
        for (int i = 0; i < values.length; i++) {
            if (num >= values[i]) {
                quotient = num / values[i];
                num %= values[i];
                
                roman += letters[i].repeat(quotient);
            }
            
        }
        
        return roman;
    }
}

File: 2. Medium/Python/Integer to Roman.py
def intToRoman(num):
    output = ''
    romans = {'M':1000,'CM':900,'D':500,'CD':400,
              'C':100, 'XC':90, 'L':50, 'XL':40,
              'X':10,  'IX':9,  'V':5,  'IV':4, 'I':1}
    
    for unit in romans.keys():
        if num >= romans[unit]:
            output += unit * (num // romans[unit])
            num = num % romans[unit]
            
    return output
